### DISCUSSION TITLE
There are plans to support both erlang and Elixir

### DISCUSSION THREAD

**Discussion by leeyisoft** [akash-akya/vix#114] [General]
There are plans to support both erlang and Elixir

**Comment by akash-akya** [comment]
Hi @leeyisoft. 
Currently, code heavily depend on Elixir macros to generate functions, documentation for all 300+ operations. And precompiled binaries depend on `cc_precompiler`.

But I guess we could cut few features to target Erlang. I'll look into it.

PS: If you don't mind working with NIF, you can just copy all `c` code and call nif from Erlang. You would still have to understand the GObject system etc, but it is doable.

**Comment by leeyisoft** [comment]
@akash-akya  Thank you very much for your reply, the vix project is very good (I'm just not familiar with Elixir, or rather: I learned erlang first, and kind of reject Elixir).

I don't have the ability to "call nif from Erlang" yet, I'm going to learn GObject

**Comment by akash-akya** [comment]
Understood. If you need any help with integration, I can assist. You should be able to reuse most of the NIF c code.

> I don't have the ability to "call nif from Erlang" yet, I'm going to learn GObject

Unfortunately, the documentation on GObject is sparse. [This](https://www.libvips.org/API/current/binding.html) might be a good starting point. I also found the libvips code itself a good place to understand the API.
